Manufacturers

elaint03 opened the pull request.

elaint03 committed:
Code exploring the BMI variable (IQR and outliers)

elaint03 committed:
This code is from the second data linkage file in the sequence (the linkage of the purchase and panel data being the first).

This code links the HFSS and NPM data to the purchases in the purchase/panel data file.

Please check this code for accuracy.

elaint03 committed:
I have renamed this file so that there is a common format for all of the linkage files I create.

I also added a line of code to delete the null variables following the merge of the panel and purchase data for 2022 and for 2023 to keep the file tidy.

elaint03 committed:
I have renamed the file "Panel purchase data linked to HFSS" to "LINKAGE Panel_purchase+HFSS 2022 and 2023"so that there is a common format for all of the linkage files I create.

I have updated the code to do a left merge of the panel_purchase data with the HFSS file, retaining only the "NPM for Period" variable from the HFSS file in the new merged file

elaint03 committed:
This file was renamed as "LINKAGE Panel_purhase+HFSS 2022 and 2023"

elaint03 committed:
Lines 48-63 added code to join names of markets/submarkets/sectors/area/extended for 2023 data

Lines 65-78 att product code and uom information - NOT COMPLETED

Lines 118-133 added code to join names of markets/submarkets/sectors/area/extended for 2023 data

elaint03 committed:
Further work to add store, prpduct and producer info - NOT COMPLETED

elaint03 committed:
SG analysis code to obtain the total monetary spend and volume of HFSS purchases for 2023, split by manufacturer.

There is a problem with the data currently as there are over 6,000 purchases which could not be linked to valid NPM/HFSS data from the NPM2024P2 file.

elaint03 committed:
Deleted code

elaint03 committed:
Deleted code

elaint03 committed:
Code to linik main panel and purchase data file with the following:

1 NPM2024P2 file
2. Product area/market/sector/submarket/extended
3. Store channel data (stores05)
4. Product information (rst_products)
5. uom data (rst_uom)

STILL TO BE LINKED
1. Price promotion code descriptions
2. HFSS regulation categories (speak to Catriona)
3. Nutrition information
Manufacturer information for 2022 and 2023

Currently an unresolve dissue with missing HFSS scores for over 6,000 purchases in 2023 due to missing look up data in the NOM2024P2 file

elaint03 committed:
Code to mereg panel and purchase data for 2022 and 2023 (first linkage file)

elaint03 committed:
Added message on line 43 to tell me where to start when I return to this file on 11/09/24

elaint03 committed:
I tidied up the code descriptors/info to help with QA

elaint03 committed:
I tidied up code descriptors to help with QA

elaint03 committed:
Code creaetd to input missing NPM and HFSS info for 6,014 purchases across 550 manufacturers.

Thre were a further 3,081 purchases across 60 products with no NPM or HFSS status that couldn't be filled from nearest purchase of same product.

Elaine has removed these.

elaint03 committed:
Took 'by' argumenyt out of rst_uom merge code on lines 102 and 229 to avoid duplication of variables.

Adjusted order of variables and some descriptor code

elaint03 committed:
I have linked Catriona's HFSS reg category file to the master purchase file for 20223 (HFSSFINAL23). I have also deleted old code that is no longer needed as a result.

clfraser commented on Code/SG Scottish manufacturer analysis Sep24.R:
```diff
@@ -0,0 +1,106 @@
+prodnames <- read_parquet("/PHI_conf/PHSci-HFSS/Kantar analysis/Working Data/npm_hfss_prod_names.parquet")
+
+#Merge master purchase dataset for 2023 with HFSS regulation category file
+SG_dataset <- dplyr::left_join(HFSSFINAL23,prodnames, by=c("prodcode" = "PRODUCT"))
```
You can avoid creating duplicate columns and then deleting them by only selecting the columns you're interested in from the prodnames dataframe. You could create a new dataframe, e.g.

prodnames_to_join <- prodnames %>%
select(PRODUCT, hfss_categories)

or do it as part of the join, as below:

```suggestion
SG_dataset <- dplyr::left_join(HFSSFINAL23,prodnames %>% select(PRODUCT, hfss_category), by=c("prodcode" = "PRODUCT"))
```

clfraser commented on Code/SG Scottish manufacturer analysis Sep24.R:
```diff
@@ -0,0 +1,106 @@
+SG_dataset <- dplyr::left_join(HFSSFINAL23,prodnames, by=c("prodcode" = "PRODUCT"))
+             
+#Tidy up file
+SG_dataset <- subset(SG_dataset, select = -c(28, 57:60, 63:65, 68))#Delete duplicate variables
```
This actually deletes both PRODUCT_LONG_DESC columns (columns 28 and 68), but since you're not using them in your analysis, it doesn't make a difference to the results.

clfraser commented on Code/SG Scottish manufacturer analysis Sep24.R:
```diff
@@ -0,0 +1,106 @@
+#Remove products which are not included in the regulation categories and which are not HFSS = 1
+SG_dataset <- SG_dataset %>% drop_na(hfss_category)#Drop non-HFSS category products (left with 765,267 purchases for 2023)
+SG_dataset <- SG_dataset[SG_dataset$HFSS.x != 0, ]#Drop products with an HFSS score of 0 within the regulation categories (left with 474,844 HFSS purchases falling into the reg categories for 2023)
+
```
If you wanted, you could do this in one go using filter from tidyverse:

```
SG_dataset <- SG_dataset %>%
     filter(!is.na(hfss_category),
            HFSS.x != 0)
```

clfraser commented on Code/SG Scottish manufacturer analysis Sep24.R:
```diff
@@ -0,0 +1,106 @@
+  group_by(Manufacturer) %>% 
+  summarise(Spend= sum(amtspent))
+
+SGTableA = SGTableA %>% 
```
You could include the 'round' in the mutate, to save a line.

```suggestion
SGTableA = SGTableA %>% 
  mutate(percent = round((Spend/sum(Spend)*100),2))
```
However, I see that the cumulative percentage only goes up to 99.52% because of earlier rounding. It's not likely to make much of a difference here, but you could calculate the percentage and cumulative percentage unrounded first, and then round both for display purposes.

The Spend, percent and cumperc columns could all be created in one expression, but I know that it makes sense as you write the code to do each calculation and check it before doing the next:

```
SGTablaA <- SG_dataset %>%
                     group_by(Manufacturer) %>%
                     summarise(Spend = sum(amtspent)) %>%
                     ungroup() %>%
                     mutate(percent = Spend/sum(Spend)*100) %>%
                     arrange(-Spend) %>%
                     mutate(cumperc = round(cumsum(percent), 2),
                                 percent = round(percent, 2))
```

clfraser commented on Code/SG Scottish manufacturer analysis Sep24.R:
```diff
@@ -0,0 +1,106 @@
+#SUMMARY STATISTICS
+
+
+SGTableA <- SG_dataset%>%        
+  group_by(Manufacturer) %>% 
+  summarise(Spend= sum(amtspent))
```
Remember to add ungroup if you group something. It doesn't make any visible difference to the dataframe, but it can cause various problems with other operations later if it remains grouped.

```suggestion
SGTableA <- SG_dataset%>%        
  group_by(Manufacturer) %>% 
  summarise(Spend= sum(amtspent)) %>%
  ungroup()
```

clfraser reviewed (approved):
This looks good. I've only looked at the manufacturer analysis file in detail. I've added some comments, but none of them are necessary to make the code work, just suggestions for how things could be done differently.

elaint03 committed:
Lines 74-110 updated to add a spend variable that takes into account the grossup factor needed to get spend at full panel level. This changes a couple of manufacturers at the bottom end and the order of the others slightly.

elaint03 committed:
LINES 74 - 78 Gross up factor included in calculation of total spend

LINES 86 onwards use variable amtspent_panel (renamed Spend) to allow for grossing up factor to be applied in summary tables

elaint03 committed:
I found an error with the code linking the HFSS regulation category file from Kantar with the master data file.

The revised code starts on row 285 to row 301 and creates two Kantar regulation category files: th first based on those rows which have a value in the extended_desc column and the second based on rows which had only NA values in the extended_desc column.

This allowed me to merge the master category file (market_sub_extend) with the two Kantar reg files separately: the first based on 3 IDS variables and the second based on 2 ID variables. I then appended them together.

The final stage was to merge this with the master file so that all HFSS purchases were correctly identified as being under the regulations or not

elaint03 committed:
I added in code to calculate the total and average number of shops per households over 2022-2023

elaint03 committed:
I added code to remove 6 panellists with English postcodes. Lines 313-314

elaint03 committed:
I revised the code here to remove the panelists with the English postcodes from the files for 2022 and 2023 separately.

elaint03 committed:
Code to create a lookup file for Kantar. Linkage ID is full postcode. This would give us a range of geographies (LA, HB, DZ, IZ, SIMD, Urban/rural).

elaint03 committed:
I corrected code to remove duplicate entries across the 2022 and 2023 nutrition files.

Please check this file is correctly doing what is expected from the written descriptions in the code file.

elaint03 committed:
Note added to say that 386 purchases/products have no associated nutritional information

elaint03 committed:
SIMD -postcode look up for Kantar updated to remove postcodes with B or C suffixes. Those with A suffixes were kept as this is where the population is weighted. The A suffix was then removed and the postcode concatenated back into one variable.

There are still 346 live postcodes without SIMD information. The SIMD - postcode lookup available appears to be out of date and there is a notice that it is due to be updated this month. The final file for Kantar would need to be refereshed with this new information before sending to Kantar.

elaint03 committed:
QA check of linkage file 1 - some code tidied up. All working.

elaint03 committed:
Code which adds promotion code descriptors between minimum and maximum provided by Kantar

e.g. tpr 0-9p a000/a009

became

tpr 0-9 a000
tpr 0-9 a001
tpr 0-9 a002
...
tpr 0-9 a009

Please check this code is correct.

elaint03 committed:
Lines 322-370

This adds the HFSS regs file to the main file. Previously some products were not being included under the reg categories because I had joined the files using different market, submarket and extended variables from the master dataset.

In this file, I added all the RF files first then used the RST 4 market, rst 4 submarket, and rst 4 extended variables.

I used AI to help with the code to link the Kantar regs file to the master file. This uses the coalesce function (lines 331-336)

elaint03 committed:
Code written to attach the two SIMD files to the master HFSS data file.

All panel members have an SIMD code, all appears ok.

elaint03 committed:
Code to derive a variable that divides products into 4 categories (NPM >=1 or 4 AND in a reg catgory, NPM< 1 AND in a reg cat, NPM >=1 or 4 and NOT in a reg cat, NPM <1 and NOT in a reg cat).

There was an issue with the RST TRADIG AREA vatiable, this was chanegd to the Sector variable

elaint03 committed:
I added code to summarise the number of purchases (HFSS in scope, HFSS exempt, and non-HFSS) made in every 4 week period between January 2022 and December 2023

elaint03 committed:
I added code to assign a promotion descriptor for the promo code "aaaa" which had been missing a descriptor in the Kantar read me file. It should now assign "TPR £10.00" to the 1,326 purchases that were previously NA

elaint03 committed:
Added code to include a promotion code variable categorised into promotion groupings for analysis

elaint03 committed:
Edited code to add a promoreg variable which allows for analysis by promotion regulation category

elaint03 committed:
Added promotion code for regulation catefory 2 - crisps and savoury snacks

elaint03 committed:
Code calculates the number of energy drinks sold as a percentage of sugar sweetened beverages (SSB) and as a percentage of total soft drinks sold in 2022 and 2023 combined
